### PR TITLE
Address book integration in frontend

### LINKED ISSUE #1420: Create a database of address names and use it in rotki
## Abstract

Rotki has various sources of data that could be used to build a user's "address book". This address book could then be used to add extra information to data that we present in the application. These sources include:
* Blockchain accounts entered by the user (data exists)
* Blockchain addressses/accounts in connected exchanges (data exists but perhaps we don't collect it in this way)
* Known addresses for popular platform smart contracts (not tokens, but contracts) (not yet supported) (e.g. ENS, UniSwap)
* 3rd party accounts that could be manually added by the user (not yet supported)
* _add more sources here_

Essentially we can use this information to provide clarity to the user when they are viewing different content within the application. For example, if they're looking at Exchange Deposits / Withdrawals, the `from address` or `to address` could be checked against the address book, and if they exist we would show the label that the user has indicated for the address as opposed to the address hash. Likewise this could be applied for transactions with known platforms/smart contracts, or incoming/outgoing transactions from/to addresses that the user has identified and labeled manually (e.g. the user might have identified & labeled accounts that belong to friends/family, a product they are paying for (rotki), etc.).

We should also determine to what extent we can integrate information from DID providers such as 3box its evolution such as cutting edge identity frameworks such as identity index (https://twitter.com/identityindex) etc.

This functionality would eventually serve as the backbone for enriching the application with "business-minded" functionality such as the identification the accounting of customers and vendors (together with the double entry work we will do), or analytical functionality by tagging addresses and transactions and thus allowing the user to analyzing things like cash flows, budgeting, etc.

EDIT:
Note: I can't imagine we can tackle this as a "one and done" issue, so I propose we keep this as a general high-level issue and open and link specific issues as we identify and work against them.

### PR DESCRIPTION
**Pull request by lukicenturi**
Closes #1420

## Checklist

- [ ] The PR modified the frontend, and updated the [user guide](https://github.com/rotki/rotki/blob/develop/docs/usage_guide.rst) to reflect the changes.

### CODE REVIEW

File: frontend/app/src/store/balances/index.ts [hunk 1]
```diff
@@ -63,38 +67,104 @@ export const useEnsNamesStore = defineStore('ensNames', () => {
...
+  const ethNameSelector = (address: string) => {
     return computed<string | null>(() => {
       if (!get(isEnsEnabled)) return null;
-      return get(ensNames)[address] ?? null;
+      return get(ethNames)[address] ?? null;
     });
   };
 
+  const updateEthAddressBookState = (
+    prefix: EthAddressBookPrefix,
```
Comment (nebolax) [inline frontend/app/src/store/balances/index.ts:99]: Note: `/global` or `/private` is a postfix, not prefix

File: frontend/app/src/router/routes.ts [hunk 2]
```diff
@@ -163,6 +163,11 @@ export const Routes = {
     icon: 'mdi-chart-line',
     text: i18n.t('navigation_menu.manage_prices').toString()
   },
+  ETH_ADDRESS_BOOK_MANAGER: {
```
Comment (kelsos) [inline frontend/app/src/router/routes.ts:166]: btw as we migrate to vue3 we should start looking into [Vitesse](https://github.com/antfu/vitesse/) for improvement ideas. an thing that I liked for example was [vite-plugin-pages](https://github.com/hannoeru/vite-plugin-pages) that makes things closer to how they work with nuxt.

**Review by kelsos — APPROVED** [review]
Looks good and works fine.

One minor thing I noticed was that I had an address with a label, I changed added an address book entry and then removed the label, but the display in the account and balances didn't change until I re-logged in.

File: frontend/app/src/components/EthNamesHint.vue [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+      </div>
+    </div>
+  </v-menu>
+</template>
+<script lang="ts">
+import { defineComponent } from '@vue/composition-api';
+import i18n from '@/i18n';
+
+const priorityList = [
+  i18n.t('eth_address_book.hint.priority.list.private_address_book').toString(),
```
[the remark this answers is not in the document]
Comment (lukicenturi) [reply frontend/app/src/components/EthNamesHint.vue:40]: I got it from the API docs https://rotki.readthedocs.io/en/latest/api.html#search-for-all-known-names-of-an-address

**Pull request merged**